storage/heap/hp_dspace.cc

```diff
@@ -234,8 +235,10 @@ static uchar *hp_allocate_variable_chunkset(HP_DATASPACE *info,
       if (last_existing_chunk) {
         /* Truncate whatever was added at the end of the existing chunkset */
         prev_chunk = last_existing_chunk;
-        curr_chunk = *((uchar **)(prev_chunk + info->offset_link));
-        *((uchar **)(prev_chunk + info->offset_link)) = nullptr;
+        memcpy(&curr_chunk, prev_chunk + info->offset_link, sizeof(uchar *));
+        memset(reinterpret_cast<uchar **>(prev_chunk + info->offset_link), 0,
```

Comment:
Very strictly speaking the nullptr representation is not required to be all zeros - but fixing this - say, on read checking if all bytes are 0, if yes, setting to nullptr - seems waste of effort here. A better fix would be to have explicit chunk pointers instead, but this is a bigger refactoring than is currently justifiable